iphone/cli/hooks/frameworks.js

```diff
@@ -397,12 +399,10 @@ class InspectFrameworksTask extends IncrementalFileTask {
 			const metadata = await fs.readJSON(this._metadataPathAndFilename);
 			for (const frameworkPath of Object.keys(metadata)) {
 				const frameworkMetadata = metadata[frameworkPath];
-				this._frameworks.set(frameworkMetadata.name, new FrameworkInfo(
-					frameworkMetadata.name,
-					frameworkMetadata.path,
-					frameworkMetadata.type,
-					new Set(frameworkMetadata.architectures)
-				));
+				this._frameworks.set(frameworkMetadata.name, new FrameworkInfo({
+					...frameworkMetadata,
+					architectures: new Set(frameworkMetadata.architectures)
+				}));
```

Comment:
I'm not sure how to fix this eslint warning.
(If we merge this as-is, then we'll get this warning in all of our PRs.)

It's a false positive and we correctly set the node version via the "package.json" as shown below. For some reason eslint is ignoring it.
https://github.com/appcelerator/titanium_mobile/blob/master/package.json#L163-L165

We can work-around it by adding the following rule to our [.eslintrc](https://github.com/appcelerator/titanium_mobile/blob/master/.eslintrc) file, but it would be a shame to duplicate our min node version in our [package.json](https://github.com/appcelerator/titanium_mobile/blob/master/package.json) here. Especially since the eslint documentation for this rule [here](https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-unsupported-features/node-builtins.md) says we shouldn't have to do this. Hmm...
```json
"rules": {
	"node/no-unsupported-features/node-builtins": ["error", {
		"version": ">=12.13.0",
		"ignores": []
	}]
},
```

Reply:
You can use Object.assign which should do the same as the spread operator

Reply:
That's true.

I guess I'm more concerned why eslint is ignoring the node version specified in the "package.json", because according to the docs it should work. Even VS Code gives me the same warning.

Reply:
I think the issue is down to having multiple `package.json` files in the repo, it looks like the plugin walks backwards from the file with the warning/error to lookup the engines.node property, as that isn't in `iphone/package.json` it's defaulting to >=8.0.0. I've just created #12965 to fix this

Reply:
@ewanharris , that makes sense. Why didn't I think of that. :stuck_out_tongue:
For now, I worked-around it by implementing it differently.